pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -1499,13 +1500,21 @@ func TestPriorityQueue_MoveAllToActiveOrBackoffQueueWithQueueingHint(t *testing.
 			expectedQ: unschedulablePods,
 		},
 		{
-			name:    "QueueHintFunction is not called when Pod is gated",
-			podInfo: setQueuedPodInfoGated(&framework.QueuedPodInfo{PodInfo: mustNewPodInfo(p), UnschedulablePlugins: sets.New("foo")}),
+			name:    "QueueHintFunction is not called when Pod is gated by SchedulingGates plugin",
+			podInfo: setQueuedPodInfoGated(&framework.QueuedPodInfo{PodInfo: mustNewPodInfo(p), UnschedulablePlugins: sets.New(names.SchedulingGates, "foo")}),
 			hint: func(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
 				return framework.Queue, fmt.Errorf("QueueingHintFn should not be called as pod is gated")
 			},
 			expectedQ: unschedulablePods,
 		},
+		{
+			name:    "QueueHintFunction is called when Pod is gated by a plugin other than SchedulingGates",
+			podInfo: setQueuedPodInfoGated(&framework.QueuedPodInfo{PodInfo: mustNewPodInfo(p), UnschedulablePlugins: sets.New("foo")}),
+			hint:    queueHintReturnQueue,
+			// FIXME: This should be backoffQ.
+			// https://github.com/kubernetes/kubernetes/issues/125538
```

Comment:
I'd remove this comment, as it potentially depends on whether the Pod went through the scheduling cycle.

Or you can leave it as a TODO, instead of FIXME.

Reply:
Prefer to remove this, we have another issue for tracking.

Reply:
I'll just remove it for now.